CMSgov/onemac
# OY2-11149: search bar on package dashboard

Story: https://qmacbis.atlassian.net/browse/OY2-11149
Endpoint: https://d14npdhsp7cyaa.cloudfront.net/

### Details

Create a search bar on the "new" package dashboard.

### Changes

- Add a search bar to the top of the package dashboard.
- Implement filtering on ID and Submitted By columns
- Implement "clear search" feature via floating X button

### Implementation Notes

- This uses the builtin "global" filtering hook in React Table. This seems slightly silly given that we are disabling this so-called "global" filter on all but two columns, but is genuinely the best-supported and most concise way to go.
- This search bar is built into our table component itself rather than the dashboard. This will also mean that the filter menu will need to built in there too. Keep this in mind for the imminent followup ticket implementing that. In addition, keep this in mind for when we inevitably need to implement a search bar for User Management - the work is already 90% done there.

### Test Plan

1. Log in as `statesystemadminactivemi`. Create at least one submission but then log out.
2. Log in as `statesubmitteractive`. Create some more submissions under both MI and VA.
3. Go to the package dashboard and check out the search bar. Verify that it matches the (newer) Figma design that is currently linked in the story.
4. Test the ID filtering feature by typing "mi" into the search bar. Verify that only packages from Michigan are listed in the table. Clear the search using the "X" button and verify that all packages are now displayed.
5. Test the submitter name filter by typing "abb". Verify that only the package(s) you created as SSA Abby Active are now displayed.
6. With or without clearing the bar, type some random characters or a string that is obviously not a name or ID number. Verify that the table body is replaced by the "no results found" state shown in the Figma.
